### ISSUE TITLE
watch-files doesnt consistently trigger

### ISSUE THREAD

**Issue by lockmeister** [Aider-AI/aider#2586]
I can't consistently get aider to trigger using `watch-files`

I've tried a few different versions including the examples in https://aider.chat/docs/usage/watch.html (I can get the examples to work).
but I usually can't get it to work. I'm using VSCode as my editor, with aider running in a terminal to the side.
For example, here in a file with a `.sh` extension:

`function to print files in current dir #!AI` then I hit ctrl-s to save it . Aider prompt refreshes automatically but there is no output.
this doesn't work either
`#AI: create a function that prints all files in current dir `
nor does
`echo the username #!AI`

Aider version: 0.67.0 ( I just upgraded to master branch `python -m pip install --upgrade git+https://github.com/Aider-AI/aider.git`)
Python version: 3.10.12
Platform: Linux-5.15.167.4-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python implementation: CPython
Virtual environment: Yes
OS: Linux 5.15.167.4-microsoft-standard-WSL2 (64bit)
Git version: git version 2.34.1

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

The trigger is `AI!`. Looks like you were using `!AI` ?

**Comment by lockmeister** [comment]
good catch, that is part of the issue. I did do it the right way for most of my testing but wrote up the issue incorrectly.
Also, I was trying to use it to edit a .txt file and I think it only works for some file extensions?

**Comment by karmajunkie** [comment]
I'm also having no luck getting it to trigger. What file extensions will it work in? I'm working in an elixir codebase, so most of mine are either `.ex` or `.exs`

**Comment by karmajunkie** [comment]
My own issue was that Elixir endings were not in the list contained in `watch.py`; #2605 fixes this. 

@lockmeister it looks like `txt` files aren't in the list either, probably because they don't have a known single-line-comment format.

**Comment by tom-doerr** [comment]
Maybe it should be clarified in the release notes that the trigger is `AI!`. Tried the last 15 mins unsuccessfully to engage it using the `# AI comments` syntax.

**Comment by lockmeister** [comment]
perhaps there could be a default trigger and also a user-defined trigger
